src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/EventBookingManager.java

```diff
@@ -873,6 +1084,55 @@ private void ensureCapacity(final IsaacEventPageDTO event, final RegisteredUserD
         }
     }
 
+    /**
+     * Helper method to ensure a batch can be booked onto an event without violating space restrictions on the event.
+     * <p>
+     * If it does an exception will be thrown if a new booking wouldn't no exception will be thrown.
+     *
+     * @param event the event the user wants to book on to
+     * @param users  the users who are trying to be booked onto the event.
+     * @throws SegueDatabaseException - if an error occurs
+     * @throws EventIsFullException   - if the event is full according to the event rules established.
+     */
+    private void ensureCapacity(final IsaacEventPageDTO event, final List<RegisteredUserDTO> users) throws
```

Comment:
Are we assuming reservations whose interval has elapsed don't count towards capacity?

Reply:
That should be the case but it seems like it isn't (EventBoonkingManager.java, L 818). I think somehow I assumed we'd run a cronjob to remove the expired reservations. If we don't do that, we need to change the query in PgEventBookings.java, L 316, to exclude expired reservations.

Reply:
We should implement one of these solutions

Reply:
Option 2 is probably the easiest but I can't recall what I was looking at when I referred to that query and I'm now lost as that line is now an empty line…

Reply:
This should be resolved in #264 by adding a Quartz job.